plugins/module_utils/foreman_helper.py

```diff
@@ -816,7 +816,12 @@ def _lookup_entity(self, identifier, entity_spec, params=None):
         try:
             if 'scope' in entity_spec:
                 for scope in entity_spec['scope']:
-                    params.update(self.scope_for(scope))
+                    params.update(self.scope_for(scope, _scope_key(resource_type, scope)))
+            if 'optional_scope' in entity_spec:
```

Comment:
this now works for everything but the main `entity`, should that also get optional scopes? I think so!

Reply:
Good call.